usr/lib/common/template.c

```diff
@@ -1896,6 +1904,9 @@ CK_RV template_validate_base_attribute(TEMPLATE *tmpl, CK_ATTRIBUTE *attr,
         break;
     case CKA_UNIQUE_ID:
         break;
+    case CKA_IBM_PROTKEY_EXTRACTABLE:
```

Comment:
CKA_IBM_PROTKEY_EXTRACTABLE is added to key objects only in key_object_set_default_attributes(), so you should only allow it on key objects. I.e. this check must be moved to key_object_validate_attribute() in key.c. Otherwise one could set attribute CKA_IBM_PROTKEY_EXTRACTABLE on non-key objects also.

Reply:
Done.